### PR TITLE
Fix repair schedule

### LINKED ISSUE #3534: SM 3.2 - SCTOOL progress percentage over 100%
##Problem
After upgrade to SM 3.2 Sctool displays progress over 100%

Status:		RUNNING
Start time:	23 Aug 23 08:12:53 UTC
Duration:	2h37m24s
Progress:	103%

```

├───────────────────────────────┼─────────────────────────────────────┼──────────┼────────────┤
│ system_auth                   │                     role_attributes │ 123%     │ 214h42m23s │
│ system_auth                   │                        role_members │ 123%     │ 214h42m53s │
│ system_auth                   │                    role_permissions │ 123%     │ 214h43m59s │
│ system_auth                   │                               roles │ 123%     │ 214h43m25s │
├───────────────────────────────┼─────────────────────────────────────┼──────────┼────────────┤
│ system_distributed_everywhere │      cdc_generation_descriptions_v2 │ 307200%  │ 7m26s      │
├───────────────────────────────┼─────────────────────────────────────┼──────────┼────────────┤
│ system_distributed            │         cdc_generation_descriptions │ 123%     │ 214h46m7s  │
│ system_distributed            │           cdc_generation_timestamps │ 123%     │ 214h45m3s  │
│ system_distributed            │         cdc_streams_descriptions_v2 │ 123%     │ 214h46m38s │
│ system_distributed            │                      service_levels │ 123%     │ 214h44m30s │
│ system_distributed            │                   view_build_status │ 123%     │ 214h45m34s │
╰───────────────────────────────┴─────────────────────────────────────┴──────────┴────────────╯
```
One table reached:
| system_distributed_everywhere |      cdc_generation_descriptions_v2 | 307200%  | 2h32m27s   |

### PR DESCRIPTION
**Pull request by Michal-Leszczynski**
This PR fixes situations when repair task created before SM 3.2 and started after upgrading to 3.2 displayed wrong progress. With those changes, it should work fine even for paused repair tasks resumed after the upgrade.

Fixes #3534

### CODE REVIEW

File: pkg/service/repair/export_test.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/scylladb/gocqlx/v2"
+)
+
+func SetLastResumableRunID(t *testing.T, s gocqlx.Session, run *Run) {
+	t.Helper()
```
[the remark this answers is not in the document]
Comment (Michal-Leszczynski) [reply pkg/service/repair/export_test.go:13]: That's the main part, but it also validates that the run isn't too old or successful.

File: pkg/service/repair/export_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2023 ScyllaDB
```
Comment (karol-kokoszka) [inline pkg/service/repair/export_test.go:1]: If I understand correctly, this file is created to have a possibility of invoking 
`(pm *dbProgressManager) setPrevRunID(ctx context.Context, ageMax time.Duration) `
outside of the `repair` package.

Integration tests are black-box ones, and they are part of `repair_test` package, so to let them to invoke what you needed, you put a wrapper over needed function and place it inside `_test` postfixed file. It means that export_test will not be part of the output binary.

It's clever, but hard to understand at the first glance.

I suggest to either:
- add the comment explaining the need for this file 
- just export `dbProgressManager` and the method you need `setPrevRunID`

**Review by karol-kokoszka — APPROVED** [review]
:+1: but please add some comment with explanation of why `export_test` is introduced.

File: pkg/service/repair/export_test.go (hunk 2 above)
Comment (Michal-Leszczynski) [reply pkg/service/repair/export_test.go]: I actually like the export approach more.
I added a commit that exports ProgressManager interface alongside with NewDBProgressManager function.
This also makes the code look cleaner and make the interface more straight forward.

**Comment by Michal-Leszczynski** [comment]
@karol-kokoszka I added two commits (one to be squashed) and addressed the `export_test` issue.

**Review by karol-kokoszka — APPROVED** [review]

**Pull request merged**